src/ctapipe/io/hdf5merger.py

```diff
@@ -267,7 +283,31 @@ def _check_can_merge(self, other):
                     f"Required node {node_path} not found in {other.filename}"
                 )
 
+    def _check_obs_ids(self, other):
+        ob_key = "/configuration/observation/observation_block"
+        # fallback for old files without ob config
+        trigger_key = "/dl1/event/subarray/trigger"
+
+        if ob_key in other.root:
+            key = ob_key
+        elif trigger_key in other.root:
+            key = trigger_key
+        else:
+            self.log.warning(
```

Comment:
Is a missing obs_id really such an innocent error that throwing a warning should be the default action?

 I feel like a `only_warn` cli option and a default to raising an `AttributeError` or something similar would be safer.

Reply:
I wasn't totally sure on what kinds of input files the merge tool works and if we really do want to require all files on which it works to have either the observation block configuration or the dl1 subarray trigger table.

However, thinking about it again, either of those should probably always be there.